c/CMakeLists.txt

```diff
@@ -35,6 +35,8 @@ add_subdirectory(vendor/nanoarrow)
 add_subdirectory(driver/common)
 add_subdirectory(driver/framework)
 
+install(FILES "${REPOSITORY_ROOT}/c/include/adbc.h" DESTINATION include/arrow-adbc)
```

Comment:
I might be misunderstanding when we want this to be installed, but in main the driver_manager CMake configuration is responsible for installing the adbc.h file; figured it made more sense in the top level config, but may be misunderstanding